### PR TITLE
Digitalocean support for Ruby app.

### PR DESCRIPTION
**Pull request by enxebre**
This is a first bit for putting Digitalocean in place for Ruby.
If you are happy with this approach, I'll improve a little bit this PR so It can get in.
Then will create follow ups PR for providing support to the other types of Apps.
Once we have two infrastructure for every type of App, analyse common parts, refactor and remove duplication.
Let me know your thoughts!

### CODE REVIEW

**Comment by enxebre** [comment]
Added some unit testing, This should be ready to review now.

**Comment by justincampbell** [comment]
This is great! I tested it out with the following Appfile:

```hcl
application {
  type = "ruby"
}

project {
  name = "hello-from-rack"
  infrastructure = "digitalocean"
}

infrastructure "digitalocean" {
  type = "digitalocean"
  flavor = "simple"
}
```

I think the only thing missing is a Terraform output for DNS name/IP address. When I `otto deploy`, there was no output for how to access the running instance. See the AWS DNS output here: https://github.com/hashicorp/otto/blob/master/builtin/app/ruby/data/aws-simple/deploy/main.tf.tpl#L51-L53

We'll also need to update documentation on the website.

**Comment by enxebre** [comment]
Cool, just added the terraform output and some docs for the Website.
Thanks.

File: builtin/foundation/consul/data/digitalocean-simple/deploy/variables.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#--------------------------------------------------------------------
+# Access Info
+#--------------------------------------------------------------------
+
+variable "api_token" {
+    description = "Access key for AWS"
```
Comment (fatroom) [inline builtin/foundation/consul/data/digitalocean-simple/deploy/variables.tf]: Isn't it should be 'Access key for DigitalOcean'?

File: builtin/foundation/consul/data/digitalocean-simple/deploy/variables.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+variable "region" {
+    description = "Region where we will operate."
+}
+
+#--------------------------------------------------------------
+# General settings
+#--------------------------------------------------------------
+
+variable "image" {
+    description = "AMI to launch with Consul"
```
Comment (fatroom) [inline builtin/foundation/consul/data/digitalocean-simple/deploy/variables.tf]: I believe DO doesn't support Amazon Machine Image... It's just image.

**Comment by enxebre** [comment]
Conflicts are because of a bindata.go file so I just created https://github.com/hashicorp/otto/pull/230
I'll rebase this once that once get merge

**Comment by mitchellh** [comment]
Thanks! 

We're going to hold off for a version or two here on merging new infrastructures. We want to make sure we flesh out the abstractions necessary for upgrading infrastructures and so on over time. This might require core changes and we don't want to be on the hook for a bunch of different infra types yet.

I'm going to keep this PR open in the mean time and we'll come back to it then. Thanks again.

**Comment by enxebre** [comment]
Make sense, I'd be happy to rebase/rework on this PR when the time comes.

**Comment by armon** [comment]
Thank you for your contribution. Sadly, we have [decommissioned Otto](https://www.hashicorp.com/blog/decommissioning-otto.html), and are no longer developing or supporting the project.

**Pull request closed without merging**